app/controllers/concerns/foreman/controller/session.rb

```diff
@@ -26,7 +26,9 @@ def update_activity_time
   end
 
   def set_activity_time
-    session[:expires_at] = Setting[:idle_timeout].minutes.from_now.to_i
+    unless session[:sso_method] == 'SSO::Jwt'
+      session[:expires_at] = Setting[:idle_timeout].minutes.from_now.to_i
```

Comment:
Can we ask the SSO if it sets the expiry time instead of hardcoding stuff for JWT here?
```ruby
def set_activity_time
  return if available_sso.sets_expiry_time?
  session[:expires_at] = Setting[:idle_timeout].minutes.from_now.to_i
end
```

Reply:
Yes, I think I can achieve this, thanks!

Reply:
@timogoebel I tried using `available_sso.sets_expiry_time?`. 
It works fine when we first authenticate the user. 
However, when we use sessions in subsequent calls the method fails. 

That is if I run `hammer auth login basic --username admin --password changeme` it works correctly.
Immediately after when I run `hammer os list ` (which will use session since we are already autheticated in the previous command), it fails because all the SSO methods (Basic, JWT, openIDConnect etc) return a false for[ `{SSO::Method}.available?`.](https://github.com/theforeman/foreman/blob/4df81848ee7bd88f5bd58f5473aa0dff74f59de0/app/services/sso.rb#L6).

 Hence we get[ `available_sso` ](https://github.com/theforeman/foreman/blob/4df81848ee7bd88f5bd58f5473aa0dff74f59de0/app/controllers/concerns/foreman/controller/authentication.rb#L4)value as `nil` when using session

Reply:
You should be able to use `get_sso_method` to get the SSO object stored in the session.